File: 9 - Recursion/Solution.cs
using System;
using System.Collections.Generic;

class Solution
{
    static void Main(String[] args)
    {
        var n = int.Parse(Console.ReadLine());

        Console.WriteLine(factorial(n));
    }

    static int factorial(int n)
    {
        if (n == 1) return 1;

        return factorial(n - 1) * n;
    }
}

File: 9 - Recursion/Solution.go
package main

import "fmt"

func factorial(n int) int {
	if n == 1 || n == 0 {
		return 1
	}
	return n * factorial(n-1)
}

func main() {
	var n int
	fmt.Scan(&n)

	fmt.Println(factorial(n))
}
